rust/lance-file/src/writer.rs

```diff
@@ -646,7 +647,7 @@ impl<M: ManifestProvider + Send + Sync> FileWriter<M> {
         Ok(())
     }
 
-    async fn write_footer(&mut self) -> Result<()> {
+    pub async fn write_footer(&mut self) -> Result<()> {
```

Comment:
same here: maybe we could just add a custom `Drop` and don't have to expose this